easy-ui-react/src/SearchNav/CTAGroup.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { ReactNode, useMemo, Fragment, ReactElement } from "react";
+import Help from "@easypost/easy-ui-icons/Help";
+import { Separator } from "./Separator";
+import { Menu } from "../Menu";
+import { Text } from "../Text";
+import { Icon } from "../Icon";
+import { UnstyledButton } from "../UnstyledButton";
+import { useInternalSearchNavContext } from "./context";
+import { classNames } from "../utilities/css";
+import { flattenChildren } from "../utilities/react";
+
+import styles from "./CTAGroup.module.scss";
+
+export type CTAGroupProps = {
+  /**
+   * The children of the <SearchNav.CTAGroup> element. Should include <SearchNav.CTAItem> elements.
+   */
+  children: ReactNode;
+};
+
+export function CTAGroup(props: CTAGroupProps) {
+  const { children } = props;
+  const { menuOverlayProps, ctaMenuSymbol } = useInternalSearchNavContext();
+
+  const items = useMemo(() => {
+    return flattenChildren(children);
+  }, [children]);
+
+  const totalItems = items.length;
+
+  return (
+    <>
+      <div className={classNames(styles.ctaGroup, styles.ctaGroupExpanded)}>
+        {items.map((item, index) => {
+          const isLastChild = index === totalItems - 1;
+          const itemEle = item as ReactElement;
+          return (
+            <Fragment key={itemEle.key}>
+              {itemEle}
+              {!isLastChild && <Separator group="cta" />}
+            </Fragment>
+          );
+        })}
+      </div>
+      <div className={classNames(styles.ctaGroup, styles.ctaGroupMenu)}>
+        <Menu>
+          <Menu.Trigger>
+            <UnstyledButton className={classNames(styles.menuBtn)}>
+              <Icon symbol={ctaMenuSymbol ? ctaMenuSymbol : Help} />
+              <Text visuallyHidden>menu</Text>
+            </UnstyledButton>
+          </Menu.Trigger>
+          <Menu.Overlay placement="bottom right" {...menuOverlayProps}>
+            <Menu.Section aria-label="Nav actions">
+              {items.map((item) => {
+                const itemEle = item as ReactElement;
+                const keyAsString = itemEle.key as string;
+                const originalKey = keyAsString?.substring(2);
+                return (
+                  <Menu.Item
+                    key={originalKey || itemEle.key}
```

Comment:
yeah so `flattenChildren` calls `Children.toArray` and i believe the latter when it has keys, modifies them and adds a predictable prefix. typically as i understand it, key modification is sort of a no-no, but unless im mistaken, i _think_ its ok here since its being leveraged as a bridge to a new ReactNode.